lib/http/headers.rb

```diff
@@ -94,7 +94,18 @@ def to_a
     #
     # @return [String]
     def inspect
-      "#<#{self.class} #{to_h.inspect}>"
+      "#<#{self.class} #{pretty_print}>"
+    end
+
+    # Returns a printable representation of headers.
+    #
+    # @param [String] separator
+    # @return [String]
+    def pretty_print(separator = ", ")
+      to_h.map do |name, value|
+        value = value.join("; ") if value.respond_to?(:join)
+        "#{name}: #{value}"
+      end.join(separator)
```

Comment:
@ixti
Sorry, I didn't get it.
Do you mean use `"\n"` separator or do you are talking about do not join multiples values of header with `"; "`?